sos/help/__init__.py

```diff
@@ -0,0 +1,303 @@
+        help_grp = parser.add_argument_group(
+            'Help Information Options',
+            'These options control what detailed information is displayed'
+        )
+        help_grp.add_argument('topic', metavar='TOPIC', default='', nargs='?',
+                              help=('name of the topic or component to show '
+                                    'help for'))
+
+    def sanitize_topic_component(self):
+        _com = self.opts.topic.split('.')[0]
+        _replace = {
+            'clean': 'cleaner',
+            'mask': 'cleaner',
+            'collect': 'collector'
+        }
+        if _com in _replace:
+            self.opts.topic = self.opts.topic.replace(_com, _replace[_com])
+
+    def execute(self):
+        if not self.opts.topic:
+            self.display_self_help()
+            sys.exit(0)
+
+        # standardize the command to the module naming pattern
+        self.sanitize_topic_component()
+
+        try:
+            klass = self.get_obj_for_topic()
+        except Exception as err:
+            print("Could not load help for '%s': %s" % (self.opts.topic, err))
+            sys.exit(1)
+
+        if klass:
+            try:
+                ht = HelpSection()
+                klass.display_help(ht)
+                ht.display()
+            except Exception as err:
+                print("Error loading help: %s" % err)
+        else:
+            print("No help section found for '%s'" % self.opts.topic)
+
+    def get_obj_for_topic(self):
+        """Based on the help topic we're after, try to smartly decide which
+        object we need to manipulate in order to get help information.
+        """
+        static_map = {
+            'report': 'SoSReport',
+            'report.plugins': 'Plugin',
+            'cleaner': 'SoSCleaner',
+            'collector': 'SoSCollector',
+            'collector.transports': 'RemoteTransport',
+            'collector.clusters': 'Cluster',
+            'policies': 'Policy'
+        }
+
+        cls = None
+
+        if self.opts.topic in static_map:
+            mod = importlib.import_module('sos.' + self.opts.topic)
+            cls = getattr(mod, static_map[self.opts.topic])
+        else:
+            _help = {
+                'report.plugins.': self._get_plugin_variant,
+                'policies.': self._get_policy_by_name,
+                'collector.transports.': self._get_collect_transport,
+                'collector.clusters.': self._get_collect_cluster,
+            }
+            for _sec in _help:
+                if self.opts.topic.startswith(_sec):
+                    cls = _help[_sec]()
+                    break
+        return cls
+
+    def _get_collect_transport(self):
+        from sos.collector.sosnode import TRANSPORTS
+        _transport = self.opts.topic.split('.')[-1]
+        if _transport in TRANSPORTS:
+            return TRANSPORTS[_transport]
+        raise Exception("No such transport '%s'" % _transport)
```

Comment:
Worth raising an equivalent exception also in `_get_collect_cluster`, `_get_plugin_variant` and `_get_policy_by_name`?

Reply:
Actually, I think removing this exception is better as that gives us our stock "No help section found" handling.